# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1395871274 14400
#      Wed Mar 26 18:01:14 2014 -0400
# Branch gui-release
# Node ID a87bdd302b1bbc27069a0b7c30b5973b37dfbd17
# Parent  e7b347dcfc635b1daab33942eb2bb8e88bde275e
set editor cursor on error line when debug_on_error is true (bug #41532)

* input.cc (get_debug_input): If caller is defined, set
curr_debug_line to the line number in the caller.

diff --git a/libinterp/corefcn/input.cc b/libinterp/corefcn/input.cc
--- a/libinterp/corefcn/input.cc
+++ b/libinterp/corefcn/input.cc
@@ -505,32 +505,33 @@ execute_in_debugger_handler (const std::
 
 static void
 get_debug_input (const std::string& prompt)
 {
   unwind_protect frame;
 
   octave_user_code *caller = octave_call_stack::caller_user_code ();
   std::string nm;
-
-  int curr_debug_line = octave_call_stack::current_line ();
+  int curr_debug_line;
 
   bool have_file = false;
 
   if (caller)
     {
       nm = caller->fcn_file_name ();
 
       if (nm.empty ())
         nm = caller->name ();
       else
         have_file = true;
+
+      curr_debug_line = octave_call_stack::caller_user_code_line ();
     }
   else
-    curr_debug_line = -1;
+    curr_debug_line = octave_call_stack::current_line ();
 
   std::ostringstream buf;
 
   if (! nm.empty ())
     {
       if (Vgud_mode)
         {
           static char ctrl_z = 'Z' & 0x1f;
